src/hotspot/cpu/aarch64/frame_aarch64.cpp

```diff
@@ -507,11 +507,11 @@ frame frame::sender_for_interpreter_frame(RegisterMap* map) const {
   intptr_t* unextended_sp = interpreter_frame_sender_sp();
   intptr_t* sender_fp = link();
 
-#if COMPILER2_OR_JVMCI
+#if COMPILER1_OR_COMPILER2
   if (map->update_map()) {
     update_map_with_saved_link(map, (intptr_t**) addr_at(link_offset));
```

Comment:
Is it correct that this is only needed when PreserveFramePointer is false, and it's harmless to do when PreserveFramePointer is true?

Reply:
Yes. I'd say rfp always has a location, but it only can contain oop if PreserveFramePointer is false.
See MacroAssembler::build_frame()/remove_frame()
https://github.com/openjdk/jdk/blob/57f4c30fb6be1da57c8fcc742b5c36d842eef397/src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp#L5710
and frame::update_map_with_saved_link()
https://github.com/openjdk/jdk/blob/57f4c30fb6be1da57c8fcc742b5c36d842eef397/src/hotspot/cpu/aarch64/frame_aarch64.inline.hpp#L451